src/main/java/HttpUrlConnection.java

```diff
@@ -1,20 +1,20 @@
-import java.io.IOException;
+import java.io.InputStream;
 import java.net.URI;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
+import java.util.Properties;
 
 public class HttpUrlConnection {
-    static String API_KEY = "f3a1b7e47c449089b21c0fd5cc904d59";
-
-    public static void main(String[] args) throws IOException, InterruptedException {
-        var client = HttpClient.newHttpClient();
-        var request = HttpRequest.newBuilder(URI.create(String.format("http://data.fixer.io/api/latest?access_key=%s", API_KEY)))
-                .header("accept", "application/json")
-                .build();
-
-        var response = client.send(request, HttpResponse.BodyHandlers.ofString());
-
-        System.out.println(response.body());
-    }
+        public String responseFromBankAPI(String arg) throws Exception {
+                Properties prop = new Properties();
+                InputStream input = getClass().getResourceAsStream("./config.properties");
+                prop.load(input);
+                HttpClient client = HttpClient.newHttpClient();
+                HttpRequest request = HttpRequest.newBuilder(URI.create(String.format(prop.getProperty("url"), arg)))
+                        .header("accept", "application/json")
+                        .build();
+                HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
Read more about spring feign, could be used something like this: https://www.baeldung.com/spring-cloud-openfeign